api/discovery/experimental.proto

```diff
@@ -68,13 +69,13 @@ message ListGraphEdgesRequest {
 }
 
 message ListGraphEdgesResponse {
-  repeated GraphEdge edges = 1;
+  repeated GraphEdge edges = 1 [(google.api.field_behavior) = REQUIRED];
   string next_page_token = 2;
 }
 
 message GraphEdge {
-  string id = 1;
-  string source = 2;
-  string target = 3;
-  string type = 4;
+  string id = 1 [(google.api.field_behavior) = REQUIRED];
```

Comment:
What is about `(buf.validate.field).string.min_len = 1,`, `(buf.validate.field).string.uuid = true,`, ...?

Reply:
The IDs here are not UUIDs but are the resource IDs (which are probably URLs), but min length makes sense